src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,125 @@
+        Entry<K, V> entry = table[index];
+        if (entry == null) {
+            table[index] = new Entry<>(key, value, null);
+            size++;
+            return;
+        }
+        Entry<K, V> currentEntry = entry;
+        while (true) {
+            if (key.equals(currentEntry.key)) {
+                currentEntry.value = value;
+                return;
+            }
+            if (currentEntry.next == null) {
+                currentEntry.next = new Entry<>(key, value, null);
+                size++;
+                return;
+            }
+            currentEntry = currentEntry.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (size == 0) {
+            return null;
+        }
+        if (key == null) {
+            return getForNullKey();
+        }
+        int hashCode = key.hashCode();
+        int index = (table.length - 1) & hashCode;
+        Entry<K, V> entry = table[index];
+        while (entry != null) {
+            if (key.equals(entry.key)) {
+                return entry.value;
+            }
+            entry = entry.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putForNullKey(V value) {
+        Entry<K, V> entry = table[0];
+        if (entry == null) {
+            table[0] = new Entry<>(null, value, null);
+            size++;
+        } else {
+            Entry<K, V> current = entry;
+            while (true) {
+                if (current.key == null) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = new Entry<>(null, value, null);
+                    size++;
+                    return;
+                }
+                current = current.next;
+            }
+        }
+    }
+
+    private V getForNullKey() {
+        Entry<K, V> entry = table[0];
+        while (entry != null) {
+            if (entry.key == null) {
+                return entry.value;
+            }
+            entry = entry.next;
+        }
+        return null;
+    }
+
+    private void resizing() {
+        size = 0;
+        Entry<K, V>[] newTable = new Entry[table.length * 2];
+        Entry<K, V>[] oldTable = table;
+        table = newTable;
+        for (int i = 0; i < oldTable.length; i++) {
+            Entry<K, V> currentEntry = oldTable[i];
+            while (currentEntry != null) {
+                put(currentEntry.key, currentEntry.value);
+                currentEntry = currentEntry.next;
+            }
+        }
+    }
+
+    private static class Entry<K, V> {
+
+        private final K key;
+        private V value;
+        private Entry<K, V> next;
+
+        public Entry(K key, V value, Entry<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
     }
```

Comment:
You should not use getters and setters in the `Entry` class since we have access to private fields of inner classes. Using getters and setters here is redundant.

Reply:
There are NO getters and setters!